matrix_client/client.py

```diff
@@ -548,7 +553,16 @@ def upload(self, content, content_type):
             )
 
     def _mkroom(self, room_id):
-        self.rooms[room_id] = Room(self, room_id)
+        room = Room(self, room_id)
+        if self._encryption:
+            try:
+                event = self.api.get_state_event(room_id, "m.room.encryption")
+                if event["content"]["algorithm"] == "m.megolm.v1.aes-sha2":
```

Comment:
Do you need to check that `event` has a `content` property, and that `event["property"]` has an `algorithm` property (or else handle a `KeyError` exception), or is there some Python magic going on here?

Reply:
Mmmmh, there are no such checks because this field is required as per https://matrix.org/speculator/spec/drafts%2Fe2e/client_server/unstable.html#id326. The SDK is generally not robust against malformed events, though your concern is still valid, especially considering how weird it is for such a method to potentially raise a `KeyError`. Maybe @non-Jedi has some thoughts?

Reply:
Yep. In absence of a strong typing system, we assume JSON follows matrix spec. Think this is fine unless the "content" or "algorithm" fields are optional.